src/core/execute.c

```diff
@@ -411,37 +468,109 @@ int exec_spawn(Unit *unit,
         if (r < 0)
                 return log_unit_error_errno(unit, r, "Failed to serialize parameters: %m");
 
-        if (fseeko(f, 0, SEEK_SET) == (off_t) -1)
-                return log_unit_error_errno(unit, errno, "Failed to reseek on serialization stream: %m");
+        rewind(f);
 
-        r = fd_cloexec(fileno(f), false);
-        if (r < 0)
-                return log_unit_error_errno(unit, r, "Failed to set O_CLOEXEC on serialization fd: %m");
+        if (workers_available) {
+                _cleanup_free_ int *fds_array = NULL;
+                size_t n_fds_array;
 
-        r = fdset_cloexec(fdset, false);
-        if (r < 0)
-                return log_unit_error_errno(unit, r, "Failed to set O_CLOEXEC on serialized fds: %m");
+                r = fdset_to_array(fdset, &fds_array);
+                if (r < 0)
+                        return log_unit_error_errno(unit, r, "Failed to convert fdset to array: %m");
+                n_fds_array = r;
 
-        r = log_level_to_string_alloc(log_get_max_level(), &log_level);
-        if (r < 0)
-                return log_unit_error_errno(unit, r, "Failed to convert log level to string: %m");
+                r = send_many_fds(unit->manager->executors_pool_socket[0], fds_array, n_fds_array, 0);
+                if (r < 0)
+                        return log_unit_error_errno(unit, r, "Failed to send serialization fd: %m");
 
-        r = fd_get_path(unit->manager->executor_fd, &executor_path);
-        if (r < 0)
-                return log_unit_error_errno(unit, r, "Failed to get executor path from fd: %m");
+                r = fd_wait_for_event(unit->manager->executors_pool_socket[0], POLLIN, 1 * USEC_PER_SEC);
+                if (r <= 0) {
+                        log_unit_debug(unit, "No response from any worker, clearing out the pool.");
```

Comment:
i think this log message should take the value of `r` into account, to no mislead users, i.e. if timeout then say the current message, if any other failure then say so. it's fine treating the two cases the same, for the rest of the code flow, but the log message should not confuse